platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -514,7 +534,9 @@ def initialize_components(self):
             from .component import ComponentONIE, ComponentSSD, ComponentBIOS, ComponentCPLD
             self._component_list.append(ComponentONIE())
             self._component_list.append(ComponentSSD())
-            self._component_list.append(ComponentBIOS())
+            # Upgrading BIOS is not supported on SN2201
+            if DeviceDataManager.get_platform_name() not in ['x86_64-nvidia_sn2201-r0']:
```

Comment:
For component, we do not need that.
there is a per-platform component capability defined in <platform_dir>/platform_components.json. We can leverage this json file.

Reply:
Keep it as it is for now as this has not been finalized yet and we do not know whether this will be supported in the near future.

Reply:
Do you mean that we might support BIOS on 2210 in future? Not sure how it is related to by comment...
can we judge BIOS capability via sysfs?

Reply:
Yes, it's possible that upgrading BIOS FW will be supported in the feature. no sysfs. currently, it is not supported just because the vendor doesn't provide us the tool to do it.
Meanwhile, current implementation, not to add ComponentBIOS to the component list, is not the best way to do it. even upgrading is not supported, fetching version can be supported and should be exposed to user.
The fix is on the way. #45

Reply:
I see. I still insist to implement something in DeviceDataManager, it is the place to handle platform specified logic.

Reply:
As disscussed offline, we will hide the logic in `DeviceDataManager` like this:
if a dedicated componentBIOS object in `DeviceDataManager`, use it. else use the default one (`ComponentBIOS`)
fix in #46